go.mod

```diff
@@ -1,19 +1,141 @@
 module kubevirt.io/containerdisks
 
-go 1.16
+go 1.17
+
+replace (
+	github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.16.5
+
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20211217221424-8779abfbd571
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20211209144617-7385dd6338e3
+
+	k8s.io/client-go => k8s.io/client-go v0.23.5
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.23.5
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20220310132336-3f90b8c54bbb
+
+	kubevirt.io/api => kubevirt.io/api v0.51.0
+	kubevirt.io/client-go => kubevirt.io/client-go v0.51.0
+	kubevirt.io/containerized-data-importer-api => kubevirt.io/containerized-data-importer-api v1.46.0
+)
 
 require (
-	github.com/containers/image/v5 v5.17.0
-	github.com/docker/distribution v2.7.1+incompatible
+	github.com/containers/image/v5 v5.20.0
+	github.com/docker/distribution v2.8.1+incompatible
 	github.com/ghodss/yaml v1.0.0
-	github.com/google/go-containerregistry v0.7.0
-	github.com/onsi/gomega v1.10.3
+	github.com/google/go-containerregistry v0.8.1-0.20220310143843-f1fa40b162a1
+	github.com/google/goexpect v0.0.0-20210430020637-ab937bf7fd6f
+	github.com/onsi/gomega v1.19.0
+	github.com/opencontainers/image-spec v1.0.3-0.20220114050600-8b9d41f48198
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.8.1
-	github.com/spf13/cobra v1.2.1
+	github.com/spf13/cobra v1.4.0
 	github.com/ulikunitz/xz v0.5.10
-	k8s.io/api v0.20.6
-	k8s.io/apimachinery v0.20.6
-	k8s.io/utils v0.0.0-20210111153108-fddb29f9d009
-	kubevirt.io/api v0.0.0-20211129173424-e2813e40f15a
+	google.golang.org/grpc v1.45.0
+	k8s.io/api v0.23.5
+	k8s.io/apimachinery v0.23.5
+	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
+	kubevirt.io/api v0.0.0-20220308214750-7df276b474a4
+	kubevirt.io/client-go v0.0.0-00010101000000-000000000000
```

Comment:
In general you can generate a client out of `kubevirt.io/api`. But also a dynamic discovery client would be good enough to create VMs.

The only thing why client-go may make sense would be the `port-forward` call ...